File: PHP/armstrong_for_n_digits.php
<?php
/*working........
success.....*/ 
echo "we will check armstrong for a number of any digits..\n";
$n=(int)readline();
$no_of_digits=strlen((string)abs($n));
 $copy_2=$n;
 $result=0;
 echo "no. of digits- $no_of_digits\n";
 while($copy_2!=0){
    $digit=$copy_2%10;
 	$result+=($digit**$no_of_digits);
 	$copy_2/=10;
 }
if ($result==$n) {
	echo "$n is an armstrong number";
} else {
    echo "$n is not an armstrong number";
}
?>

File: C/armstrong_for_n_digits.c
#include <stdio.h>//working...
#include <math.h>//success....
int main(){
	printf("enter a number of any no. of digits\nwe will check if its armstrong or not..");
	int n;
	scanf("%d",&n);
	int copy,f=0;
	int digit,copy2;
	float result=0.0;
	for(copy2=n;copy2!=0;++f){
		copy2/=10;
	}
	for(copy=n;copy!=0;copy/=10){
		digit=copy%10;
		result+=pow(digit,f);
	}
	if((int)(result)==n)
		printf("%d is an armstrong number\n",n );
	else
		printf("%d is not an armstrong number\n",n );
}